api/src/routing.js

```diff
@@ -12,15 +12,21 @@ const prometheusMiddleware = require('prometheus-api-metrics');
 const rateLimiterMiddleware = require('./middleware/rate-limiter');
 const logger = require('./logger');
 const isClientHuman = require('./is-client-human');
-const target = 'http://' + environment.host + ':' + environment.port;
-const proxy = require('http-proxy').createProxyServer({ target: target });
+
+const port = typeof environment.port !== 'undefined' && environment.port !== null ? `:${environment.port}` : '';
+const target = `${environment.protocol}//${environment.host}${port}`;
+const proxy = require('http-proxy').createProxyServer({
+  target: target, changeOrigin: environment?.proxies?.default?.changeOrigin ?? false 
+});
 const proxyForAuth = require('http-proxy').createProxyServer({
   target: target,
   selfHandleResponse: true,
+  changeOrigin: environment?.proxies?.auth?.changeOrigin ?? false
```

Comment:
I think it'd be cleaner to remove the `?? false` and just make sure the value is always defined. I think you just need to provide a default in the unit testing setup. That simplifies this significantly, to

```suggestion
  changeOrigin: environment.proxies.auth.changeOrigin
```